katsdpcontroller/generator.py

```diff
@@ -527,14 +527,23 @@ def get_baseline_index(a1, a2):
         g.graph["static_sensors"].add(ss)
 
     bw_scale = stream.adc_sample_rate / _MAX_ADC_SAMPLE_RATE
-    # * 4 is for 2 polarisations, each with real+complex
-    batch_size = len(acv.antennas) * stream.n_chans_per_endpoint * 4 * acv.bits_per_sample // 8
+    # * 2 is for real+complex
+    batch_size = (
+        len(acv.src_streams) * acv.n_spectra_per_heap
+        * stream.n_chans_per_endpoint * 4 * acv.bits_per_sample // 8
+    )
```

Comment:
Comment doesn't seem to match up with the actual assignment.

Reply:
Thanks - I assume you're referring to the `* 4` instead of the `* 2`. I've fixed it, and now memory usage is 60-70% of allocated, which feels comfortable.